openpype/hosts/maya/api/plugin.py

```diff
@@ -143,15 +144,52 @@ def load(
         assert os.path.exists(self.fname), "%s does not exist." % self.fname
 
         asset = context['asset']
+        subset = context['subset']
+        settings = get_project_settings(context['project']['name'])
+        custom_naming = settings['maya']['load']['reference_loader']
         loaded_containers = []
 
+        if not custom_naming['namespace']:
+            raise LoadError("No namespace specified in "
+                            "Maya ReferenceLoader settings")
+        elif not custom_naming['group_name']:
+            raise LoadError("No group name specified in "
+                            "Maya ReferenceLoader settings")
+
+        custom_namespace = custom_naming['namespace'].format(
+            asset_name=asset['name'],
+            asset_type=asset['type'],
+            subset=subset['name'],
+            family=(
+                subset['data'].get('family') or
+                subset['data']['families'][0]
+            )
+        )
+
+        custom_group_name = custom_naming['group_name'].format(
+            asset_name=asset['name'],
+            asset_type=asset['type'],
+            subset=subset['name'],
+            family=(
+                subset['data'].get('family') or
+                subset['data']['families'][0]
+            )
+        )
```

Comment:
Not saying this is better - just wondering whether we should do this to avoid defining the formatting data twice. I personally don't mind the current one either.
```suggestion

        formatting_data = {
            "asset_name": asset["name"],
            "asset_type": asset["type"],
            "subset": subset["name"],
            "family": (
                subset['data'].get('family') or
                subset['data']['families'][0]
            )
        }
        custom_namespace = custom_naming['namespace'].format(
            **formatting_data
        )
        custom_group_name = custom_naming['group_name'].format(
            **formatting_data
        )
```

Reply:
Yeah, nice. @friquette you can accept the suggestion here if you want to.

Reply:
Yes this is a good idea, I will implement it !